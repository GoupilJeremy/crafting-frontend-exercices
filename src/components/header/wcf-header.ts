import css from './wcf-header.scss';

const template = document.createElement('template');
template.innerHTML = `
<style>${css}</style>
<div id="header">
    <h3 class="header__title">Web Components Bets Application</h3>
</div>
`;

export class Header extends HTMLElement {
    constructor() {
        super();

        this.attachShadow({ mode: 'open' })
            .appendChild(template.content.cloneNode(true));
    }
}