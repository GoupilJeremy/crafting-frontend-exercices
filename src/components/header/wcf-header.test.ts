import { Header } from "./wcf-header";

it('should return title header', () => {
    const header = new Header();
    expect(header.shadowRoot?.querySelector('#header')).toMatchSnapshot()
});