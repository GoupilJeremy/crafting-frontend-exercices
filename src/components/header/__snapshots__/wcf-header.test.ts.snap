// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return title header 1`] = `
<div
  id="header"
>
  
    
  <h3>
    Web Components Bets Application
  </h3>
  

</div>
`;
